docs/source/api_reference/data_format.rst

```diff
@@ -0,0 +1,273 @@
+.. _data_format:
+
+File Format Specifications
+==========================
+
+.. toctree::
+    :maxdepth: 1
+    :hidden: false
+
+    Overview
+    Introduction
+    Description
+    Instructions
+    Illustration
+
+Overview
+--------
+This document has the following content:
+
+- Introduction: What is a ``.ts`` file, why and when to use it.
+- Description: What are the individual components of a ``.ts`` file.
+- Instructions: How to create your own ``.ts`` file.
+- Illustrations: A running example to tie up above sections.
+
+Introduction
+------------
+This document formalizes string identifiers used in ``.ts`` file format.
+Encoded in ``utf-8``, ``.ts`` files stores time-series dataset and its corresponding metadata and can be
+opened via any basic editor like notepad for visual inspection.
+String identifiers refer to strings beginning with ``@`` in the file.
+
+``.ts`` files contains information blocks in the following order:
+
+1. All the comments to describe the dataset. (begins with ``#`` in the file)
+2. All the String Identifiers to give information about the metadata (begins with ``@`` in the file).
```

Comment:
"a metadata block. (now describe the block high-level) (and reference the detail description of the block"

Reply:
How about:
```
2. A metadata block.
      All lines in this block begin with ``@`` and each line has one string identifier and corresponding value.
      See :ref:`subsection on metadata block <metadata description>` for more details.
```

Reply:
better, but it does not fully spell out how lines look like.